pandas-stubs/core/indexes/base.pyi

```diff
@@ -430,12 +431,12 @@ class Index(IndexOpsMixin[S1], ElementOpsMixin[S1]):
     ) -> np_1darray[np.intp]: ...
     def reindex(
         self,
-        target,
-        method: ReindexMethod | None = ...,
-        level=...,
-        limit=...,
-        tolerance=...,
-    ): ...
+        target: Iterable[Any],
```

Comment:
the docs just say "iterable"

https://github.com/pandas-dev/pandas/blob/b015a3bfb49ec98eec2635d93ce3ebb8d5dad0c2/pandas/core/indexes/base.py#L4116